packages/fxa-content-server/app/scripts/views/pair/unsupported.js

```diff
@@ -22,21 +23,38 @@ class PairUnsupportedView extends FormView {
 
   setInitialContext(context) {
     const uap = this.getUserAgent();
+    const isFirefox = uap.isFirefox();
+    const isMobile = uap.isAndroid() || uap.isIos();
     // Assume the user is on non-Firefox desktop in this case.
-    const isDesktopNonFirefox =
-      !uap.isFirefox() && !uap.isAndroid() && !uap.isIos();
+    const isDesktopNonFirefox = !isFirefox && !isMobile;
+    const isSystemCameraUrl = this.getHashParams(['channel_id']).channel_id;
```

Comment:
Not sure if this is the best way to flag this. The `channel_id` is on both desktop and mobile pages. Mobile does have a `channel_key` param that might be used instead.

Also on mobile, I think the we might be able to check that the user has opened in Safari and no Firefox iOS using user agents.

Reply:
Thanks for calling this out! Originally I was checking for `channel_id` and `channel_key`, but I couldn't seem to get the `channel_key` check to work when `channel_id` was present for some reason and thought we wouldn't see either on desktop. Is checking for these two values sufficient, or should we check for both values plus `isMobile`?

Reply:
I think it is fine to check for all the conditions (channel_id, channel_key, isMobile). I think we still might have to tweak things but the messaging is generic ish enough that it should be ok.